Marlin/src/lcd/dogm/u8g_dev_st7920_128_64_sw_spi.cpp

```diff
@@ -0,0 +1,303 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * Please, note: 
+ *  We need our custom implementation for Software SPI, as the default implementation 
+ * of U8GLIB, when running in an ARM based board, is too fast and the display will not
+ * recognize commands and/or data at such speeds. This implementation autoderives the
+ * required delays to get the maximum possible performance by using the F_CPU macro that
+ * specifies the CPU speed. According to the ST7920 datasheet, the maximum SCLK is 1MHz.
+ */
+ 
+#ifndef ULCDST7920_SWSPI_H
+#define ULCDST7920_SWSPI_H
+
+#include "../../Marlin.h"
```

Comment:
This should only include `../../inc/MarlinConfigPre.h` before the `ENABLED` condition, and then afterward include only the minimal headers needed for this `.h` file. Don't include headers that are only needed by files that include this `.h` file. Instead, include those headers in the `.cpp` files where needed.

Note that with latest conventions, code between the outer-most conditions doesn't need to be indented, so the bulk of this file can move left 2 spaces.

Reply:
I'll do the changes .. (Y)